### ISSUE TITLE
Pandas failing on Debian big-endian builds

### ISSUE THREAD

**Issue by juricast** [pydata/pandas#10425]
Hello,

Package pandas is failing to build from source on Debian big-endian builds with following message:

<pre>
======================================================================
FAIL: test_to_records (pandas.tests.test_categorical.TestCategoricalAsBlock)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/«BUILDDIR»/pandas-0.15.1+git125-ge463818/debian/tmp/usr/lib/python2.7/dist-packages/pandas/tests/test_categorical.py", line 2393, in test_to_records
    tm.assert_almost_equal(result,expected)
  File "das/src/testing.pyx", line 58, in pandas._testing.assert_almost_equal (pandas/src/testing.c:2745)
  File "das/src/testing.pyx", line 93, in pandas._testing.assert_almost_equal (pandas/src/testing.c:1830)
  File "das/src/testing.pyx", line 143, in pandas._testing.assert_almost_equal (pandas/src/testing.c:2618)
AssertionError: (0L, 'a') != (0L, 'a')

----------------------------------------------------------------------
</pre>

https://buildd.debian.org/status/package.php?p=pandas&suite=sid

Looking at a upstream source and testing it on big-endian build I found that this problem stops occurring after commit  7eb5668

Difference between this commit and Debian version is big. 
I am new to pandas and I did not had much luck in identifying minimal change set needed to patch Debian version of package to work with big-endian.

I have seen TST: better testing for test_where (GH8997) #9015 and that did not help much.

Could someone please take a look at this?
Help is welcomed.

Thank you!

Regards,
Jurica

**Comment by jreback** [comment]
@juricast 

this is from little-endian, can you post what a big-endian does?
```
In [1]: df = DataFrame(list('abc'), dtype='category')

In [2]: df.to_records()
Out[2]: 
rec.array([(0, 'a'), (1, 'b'), (2, 'c')], 
      dtype=[('index', '<i8'), ('0', 'O')])
```

**Comment by jreback** [comment]
I suspect that test would pass with the byteorder as '=' rather than specified as '<' (little-endian).
The test is comparing incorrect with a specific byte ordered int, rather than a native ordering.

```
np.rec.array([(0, 'a'), (1, 'b'), (2, 'c')],
                                dtype=[('index', '=i8'), ('0', 'O')])
```

**Comment by juricast** [comment]
Hello,

Changing '<' to '=' on big-endian does resolve a problem.
I will build package again with this change on both endians and post results here.

Regards,
Jurica

**Comment by jreback** [comment]
gr8!

would you submit a pull-request with these changes?

**Comment by juricast** [comment]
I will close this issue and submit pull-request as soon as I have build results on both endians.
Thank you!

Regards,
Jurica

**Comment by jreback** [comment]
just submit a PR, when I merge it will close the issue automatically. thanks.

[Here](http://pandas.pydata.org/pandas-docs/stable/contributing.html) is our contributing doc

**Comment by jreback** [comment]
closed by #10438
